pkg/discovery/helper.go

```diff
@@ -193,7 +196,8 @@ func (h *helper) Refresh() error {
 
 		for _, resource := range resourceGroup.APIResources {
 			gvr := gv.WithResource(resource.Name)
-			gvk := gv.WithKind(resource.Kind)
+			// gvk kind is lowercased for consistency of hits in KindFor()
+			gvk := gv.WithKind(strings.ToLower(resource.Kind))
```

Comment:
This and related changes fixed the unit tests. Will have to try a run live in a real environment to confirm everything is working.